pkg/option/config.go

```diff
@@ -1959,6 +1947,34 @@ func (c *DaemonConfig) Populate() {
 	c.AwsReleaseExcessIps = viper.GetBool(AwsReleaseExcessIps)
 }
 
+func (c *DaemonConfig) populateNodePortRange() error {
+	nodePortRange := viper.GetStringSlice(NodePortRange)
+	if len(nodePortRange) == 2 {
+		var err error
+
+		c.NodePortMin, err = strconv.Atoi(nodePortRange[0])
+		if err != nil {
+			return errors.Errorf("Unable to parse min port value for NodePort range: %s", err.Error())
+		}
+		c.NodePortMax, err = strconv.Atoi(nodePortRange[1])
+		if err != nil {
+			return errors.Errorf("Unable to parse max port value for NodePort range: %s", err.Error())
+		}
+		if c.NodePortMax <= c.NodePortMin {
+			return errors.New("NodePort range min port must be smaller than max port!")
+		}
+	} else {
+		// TODO(ctz): This check allows NodePortRange to be passed as explicity
+		// empty, which the pkg/envoy_test.go expects. Should that be the case
+		// or does pkg/envoy_test.go need to be modified?
+		if len(nodePortRange) != 0 {
```

Comment:
nit: this `if` condition can be folded into the `else` check above.

Reply:
Fixed: cd702be